### PR TITLE
Main UI URL is hard-coded to home.myopenhab.org

### PR DESCRIPTION
**Pull request by lsafelix75**
Refer to this issue and this is a temporary workaround until a full solution is proposed. 
https://github.com/openhab/openhab-android/issues/3147

Signed-off-by: Felix Lo <lsafelix75@gmail.com>

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/activity/Oh3UiWebViewFragment.kt [hunk 1]
```diff
@@ -28,9 +28,9 @@ class Oh3UiWebViewFragment : AbstractWebViewFragment() {
     override val shortcutAction = MainActivity.ACTION_OH3_UI_SELECTED
 
     override fun modifyUrl(orig: HttpUrl): HttpUrl {
-        if (orig.host == "myopenhab.org") {
+        if (orig.host.substringBefore('.') != "home") {
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/activity/Oh3UiWebViewFragment.kt:31]:
```suggestion
        if (!orig.host.startsWith("home.")) {
```

**Comment by mueller-ma** [comment]
I'm not sure about this workaround. There could be setups that currently work.

Maybe it's better to add a server-specific URL setting, under advanced? It should be pre-filled for myopenhab.org users.

**Comment by mueller-ma** [comment]
@maniac103 WDYT here? Add this workaround or add a new setting to override the Main UI url? If latter is done, something similar could be done for #3260.

**Comment by maniac103** [comment]
I don't know the logic in openhab-cloud too well, but if the 'home' prefix is indeed hardcoded there as implied by https://github.com/openhab/openhab-cloud/issues/332 I think this workaround should be fine? Maybe it should be extended to not prepend 'home' if the hostname starts with 'remote.', though, so it matches the openhab-cloud logic.

**Comment by mueller-ma** [comment]
👍 

@lsafelix75 Can you implement the "hostname doesn't start with remote." check?

**Comment by lsafelix75** [comment]
> 👍
> 
> @lsafelix75 Can you implement the "hostname doesn't start with remote." check?

The problem is we need to create a link URL when user clicks on "it", it will be proxied to openhab instance. So, may be instead of hardcoded to home. we hardcode it to remote. because both will enter the same logic in openhabcloud app.js below:

 ```
// If host matches names for full /* proxying, go ahead and just proxy it.
    if (host.indexOf('remote.') === 0 || host.indexOf('home.') === 0) {
        //make sure this was not set by another server
        if (req.url.indexOf('/remote') != 0) {
            req.url = '/remote' + req.url;
        }
    }
```

**Comment by dastrix80** [comment]
When is this being fixed and why is it closed?

**Pull request closed without merging**